simple/templates/podmonitor.yaml

```diff
@@ -0,0 +1,28 @@
+{{- range $podMonitorName, $ref := .Values.podMonitors }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: {{ $podMonitorName }}
+  labels:
+    release: prometheus-operator
+  {{- if hasKey $ref "labels" }}
+    {{- range $key, $value := $ref.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+spec:
+  namespaceSelector:
+    matchNames:
+    {{- range $namespace := $ref.namespaceSelector }}
+      - {{ $namespace }}
+    {{- end }}
+  selector:
+    matchLabels:
+      {{- range $key, $value := $ref.selector.matchLabels }}
+      {{ $key }}: {{ $value | quote }}
+      {{- end }}
+  podMetricsEndpoints:
```

Comment:
podMetricsEndpoints not support only port field, it also support path/interval...etc important setting to customize different use case, so maybe you can change to use toYaml function instead

Reply:
ok